ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -397,6 +400,955 @@ objects:
+              - IpProtocol: tcp
+                ToPort: 3130
+                FromPort: 3130
+                CidrIp: !Ref AllowedProxyCidr
+              - IpProtocol: tcp
+                ToPort: 19531
+                FromPort: 19531
+                CidrIp: !Ref AllowedProxyCidr
+              VpcId: !Ref VpcId
+
+          ProxyInstance:
+            Type: AWS::EC2::Instance
+            Properties:
+              ImageId: !Ref RhcosAmi
+              IamInstanceProfile: !Ref ProxyInstanceProfile
+              InstanceType: "i3.large"
+              NetworkInterfaces:
+              - AssociatePublicIpAddress: "true"
+                DeviceIndex: "0"
+                GroupSet:
+                - !Ref "ProxySecurityGroup"
+                - !Ref "MasterSecurityGroupId"
+                SubnetId: !Ref "PublicSubnet"
+              UserData:
+                Fn::Base64: !Sub
+                - '{"ignition":{"config":{"replace":{"source":"\${S3Loc}","verification":{}}},"timeouts":{},"version":"2.1.0"},"networkd":{},"passwd":{},"storage":{},"systemd":{}}'
+                - {
+                  S3Loc: !Ref ProxyIgnitionLocation
+                }
+
+          RegisterProxyApiTarget:
+            Condition: DoRegistration
+            Type: Custom::NLBRegister
+            Properties:
+              ServiceToken: !Ref RegisterNlbIpTargetsLambdaArn
+              TargetArn: !Ref ExternalApiTargetGroupArn
+              TargetIp: !GetAtt ProxyInstance.PrivateIp
+
+          RegisterProxyInternalApiTarget:
+            Condition: DoRegistration
+            Type: Custom::NLBRegister
+            Properties:
+              ServiceToken: !Ref RegisterNlbIpTargetsLambdaArn
+              TargetArn: !Ref InternalApiTargetGroupArn
+              TargetIp: !GetAtt ProxyInstance.PrivateIp
+
+          RegisterProxyInternalServiceTarget:
+            Condition: DoRegistration
+            Type: Custom::NLBRegister
+            Properties:
+              ServiceToken: !Ref RegisterNlbIpTargetsLambdaArn
+              TargetArn: !Ref InternalServiceTargetGroupArn
+              TargetIp: !GetAtt ProxyInstance.PrivateIp
+
+          ProxyRecord:
+            Condition: DoDns
+            Type: AWS::Route53::RecordSet
+            Properties:
+              HostedZoneId: !Ref PrivateHostedZoneId
+              Name: !Join [".", ["squid", !Ref PrivateHostedZoneName]]
+              ResourceRecords:
+              - !GetAtt ProxyInstance.PublicIp
+              TTL: 60
+              Type: A
+
+          ProxyAliasRecord:
+            Condition: DoDns
+            Type: AWS::Route53::RecordSet
+            Properties:
+              Name: !Join [".", ["squid-lb", !Ref PrivateHostedZoneName]]
+              HostedZoneId: !Ref PrivateHostedZoneId
+              Type: A
+              AliasTarget:
+                DNSName: !GetAtt LoadBalancer.DNSName
+                HostedZoneId: !GetAtt LoadBalancer.CanonicalHostedZoneID
+
+          LoadBalancer:
```

Comment:
TODO: use the internal load balancer already created in `02_cluster_infra.yaml` in follow up PR

Reply:
@ewolinetz probably better to have the TODO in the code. Otherwise, add it as a [ ] in https://github.com/openshift/release/pull/5308

Reply:
i can remove that code from this PR, it shouldn't affect functionality we just need the `ProxyRecord`